File: src/core/manifest.js
/**
 * Returns true if the entry should be processed (not deleted or errored during dedupe).
 * When dedupe was skipped, the dedupe property is undefined, so the entry is kept.
 *
 * @param {{ dedupe?: { status: string } }} entry
 * @returns {boolean}
 */
export function canKeep(entry) {
  return entry.dedupe?.status !== "delete" && entry.dedupe?.status !== "error";
}

File: packages/core/src/pixelkasten/core/manifest.py
"""Manifest helpers for the unified pipeline."""

from pixelkasten.core.types import DedupeResult, ManifestEntry, Status


def can_keep(entry: ManifestEntry) -> bool:
    """
    Returns True if the entry should be processed (not deleted or errored during dedupe).

    When dedupe was skipped, the dedupe field is None, so the entry is kept.
    """
    if entry.dedupe is None:
        return True
    return entry.dedupe.result != DedupeResult.DELETE and entry.dedupe.status != Status.ERROR
